### ISSUE TITLE
Database/Schema/Converter.php warnings in logs

### ISSUE THREAD

**Issue by rutkas** [espocrm/espocrm#852]
I have noticed such warning in espo log files, any ideas why they are produced ?

```
[2018-03-23 14:35:35] Espo.WARNING: E_WARNING: array_key_exists() expects parameter 2 to be array, null given {"code":2,"message":"array_key_exists(
) expects parameter 2 to be array, null given","file":"/var/www/site/htdocs/application/Espo/Core/Utils/Database/Schema/Converter.ph
p","line":462,"context":{"entityList":["PurchaseDev"],"ormMeta":{"ActionHistoryRecord":{"fields":{"id":{"dbType":"varchar","len":24,"type":"id"},"na
me":{"type":"varchar","notStorable":true},"deleted":{"type":"bool","default":false},"number":{"type":"int","autoincrement":true,"unique":true,"index
":true,"len":11},"data":{"type":"jsonObject"},"action":{"type":"varchar","len":255},"createdAt":{"type":"datetime","notNull":false},"ipAddress":{"ty
pe":"varchar","len":"39"},"targetId":{"dbType":"varchar","len":24,"type":"foreignId","index":"target","notNull":false},"targetType":{"type":"foreign
Type","notNull":false,"index":"target","len":100,"dbType":"varchar"},"targetName":{"type":"varchar","notStorable":true,"relation":"target","isParent
Name":true},"userId":{"dbType":"varchar","len":24,"type":"foreignId","index":true,"notNull":false},"userName":{"type":"foreign","notStorable":false,
"relation":"user","foreign":["firstName"," ","lastName"]},"authTokenId":{"dbType":"varchar","len":24,"type":"foreignId","index":true,"notNull":false
},"authTokenName":{"type":"foreign","notStorable":false,"relation":"authToken","foreign":"id"}},"relations":{"authToken":{"type":"belongsTo","entity
":"AuthToken","key":"authTokenId","foreignKey":"id","foreign":"actionHistoryRecords"},"target":{"type":"belongsToParent","key":"targetId"},"user":{"
type":"belongsTo","entity":"User","key":"userId","foreignKey":"id","foreign":null}},"collection":{"orderBy":"number","order":"DESC"},"params":{"engi
```
... long long text, much longer..

**Comment by tmachyshyn** [comment]
It seems like some definition error can be in metadata of your custom entity "PurchaseDev". Please print or attach a file of your entity "PurchaseDev" defined in metadata under entityDefs.

**Comment by rutkas** [comment]
Thank you for the tip. I have examined my data structure, and found that I do not have PurchaseDev object/entity/table at all, I think it was deleted long time ago.. I even do not remeber it.. But link and link-multiple field was left on it in another entity. Also it was a link on it in "Meeting.json". I do not know why these links were left in metadata... I think if I deleted an object in Espo admin panel - all links to it  should be erased too.. I have removed that old links/fields and looks like warning has gone now..
